docs: vee-validate plugin documentation (#133)
vuelidate/formvuelatte: index.md

])
```

### Vee-Validate Plugin

> [Lookup Plugin's Repo](https://github.com/vuelidate/formvuelate-plugin-vee-validate).

Reviewer: ```suggestion
[Vee-Validate Plugin's Repo](https://github.com/vuelidate/formvuelate-plugin-vee-validate).
```

```js
import { SchemaFormFactory } from 'formvuelate'
import VeeValidatePlugin from '@formvuelate/plugin-vee-validate'

const SchemaFormWithValidation = SchemaFormFactory([
  VeeValidatePlugin({
    // plugin configuration here
  })
])
```

Now that the component is created, you can register it and use it in your template:

```html
<template>
  <div id="app">
    <SchemaFormWithValidation
      :schema="mySchema"
      v-model="myData"
    />
  </div>
</template>

<script>
export default {
  components: {
    SchemaFormWithValidation
  },
  setup () {
    [...]
  }
}
</script>
```

#### Component Requirements

Your components will receive the validation state via `validation` prop which contains the error messages and the `meta` information exposed by vee-validate, the validation prop contains the following properties/methods:

```js
{
  errorMessage, // The first error message for that field
  errors, // All error messages for that field
  meta, // A field meta object
  setDirty, // Sets the meta `dirty` flag
  setTouched, // Sets the meta `touched` flag
}
```

> You can opt-in to any of these properties or to the entire `validation` object, here is a simple `FormText` component that accepts the `validation` object as a prop:

Reviewer: ```suggestion
You can opt-in to any of these properties or to the entire `validation` object. Here is an example `FormText` component that accepts the `validation` object as a prop:
```

```html
<template>
  <div>
    <input
      :value="modelValue"
      @input="update($event.target.value)"
    >
    <span>{{ validation.errorMessage }}</span>
  </div>
</template>

<script>
export default {
  props: {
    // other props
    modelValue: {
      required: true
    },
    validation: {
      type: Object,
      default: () => ({})
    }
  },
  methods: {
    update (value) {
      this.$emit('update:modelValue', value)
    }
  }
}
</script>
```

> Whenever the `modelValue` is updated the field will be validated immediately. If you don't want that and want your validations to be less aggressive, then you can show the error message when the field is `touched` which is set after the field is blurred and you can set it using `setTouched` on the validation object.

Reviewer: ```suggestion
Whenever the `modelValue` is updated the field will be validated immediately. 

If we want the validations not to be immediate, or lazy, we can show the error message when the field is `touched`. A field is considered `touched` after the field loses focus. 

In this case, we set it using the validation object's `setTouched` method as shown in the following example.
```

```html
<template>
  <div>
    <input
      :value="modelValue"
      @input="update($event.target.value)"
      @blur="onBlur"
    >
    <span v-if="validation.touched">{{ validation.errorMessage }}</span>
  </div>
</template>

<script>
export default {
  props: {
    // ...
  },
  methods: {
    update (value) {
      this.$emit('update:modelValue', value)
    },
    onBlur() {
      this.validation.setTouched(true);
    }
  }
}
</script>
```

#### Configuration

> The `VeeValidatePlugin` takes one parameter which is a configuration object, here are the properties that you can use.

Reviewer: ```suggestion
The `VeeValidatePlugin` accepts one parameter, a configuration object. Let's look at the properties that we can use.
```

##### mapProps

> If you are using 3rd party components and cannot modify their definition to accept the `validation` object, you can use the `mapProps` configuration to map the `validation` object to another property or multiple properties that are accepted by your component.

Reviewer: ```suggestion

::: warning Important
This refers to Vee-validate's `mapProps`, and not to the `mapProps` property exposed by `LookupPlugin`. 
:::

If you are using 3rd party components and cannot modify their definition to accept the `validation` object, you can use the `mapProps` configuration to map the `validation` object to another property or multiple properties that are accepted by your component.
```
